### PR TITLE
Add game logic

### PR DESCRIPTION
**Pull request by pbkabali**
### Milestone 3 of Tic Tac Toe game design

- Add class for managing game board
- Add class for managing players
- Create methods to check winning and draw board
- Create methods to take user input and display output

### CODE REVIEW

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 
### HELLO @pbkabali AND @jssarrazolaa VERY NICE WORK YOU HAVE MET ALMOST ALL THE REQUIREMENTS JUST SOME FEW CHANGES TO BE MADE KINDLY CHECK BELOW :arrow_down: 

- [x] There some rubocop errors in your project. Kindly fix all this errors before submitting your project for another review.

- [x] Kindly separate your classes to diffrent files. This will make your project more cleaner.

Once you have done this changes kindly submit your work for another review.

### you can reach me via slack on @peterrobert

### HAPPY CODING :computer:

**Comment by pbkabali** [comment]
Hello @peterrobert, thank you for your review. We have worked on the feedback and submitted another review-request.

File: bin/main.rb
Comment (bertil291utn) [inline bin/main.rb:13]: ## Not in the main file 
-  This method should come from the `game_logic.rb` file, not in the main file
  - [ ] Thus, please move to `game_logic.rb` file,

File: lib/game_logic.rb
Comment (bertil291utn) [inline lib/game_logic.rb]: ## Only one action

- The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing). Student's tends to create long methods that are doing multiple steps of the game process. (screenshot from Pathwright)
  - [ ] Please remove in another function

*Example:*

```ruby
 def winner_pattern
    board_patterns = [[@board[0], @board[1], @board[2]],
                      [@board[3], @board[4], @board[5]],
                      [@board[6], @board[7], @board[8]],
                      [@board[0], @board[3], @board[6]],
                      [@board[1], @board[4], @board[7]],
                      [@board[2], @board[5], @board[8]],
                      [@board[0], @board[4], @board[8]],
                      [@board[2], @board[4], @board[6]]]
end
```

**Review by bertil291utn — CHANGES_REQUESTED** [review]
# **Status:** Changes requested 🟠 
<p align="center"> image <br><strong>Not yet</strong> </p>

# Hey guys, you've been doing a great job, and also I hope you enjoyed coding.

### Your work is amazing you master it GitHub Flow you have a professional README file. 

However, in order to accomplish all the requirements, you required to do these changes, listed below:

### Summary

-  Don't forget to work with inline comments below this comment box.

***

Guys work in the requested changes above and request another review.
Good luck. 

> “Never stop, never conform until what is good becomes even better, and what’s better becomes excellent.”

File: lib/game_logic.rb
Comment (pbkabali) [reply lib/game_logic.rb]: Thanks @bertil291utn. We have implemented this

File: bin/main.rb
Comment (pbkabali) [reply bin/main.rb:13]: We had to implement this in the main file since the instructions are to have `puts` and `gets` only in this file and nowhere else

**Comment by krys2fa** [comment]
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this project approved.  🥂

You can now merge this branch into the master branch and go on with your next project.

💡  Do not forget to submit your project completion form after merging this branch.

All the best team,
Chris

**Comment by pbkabali** [comment]
Thank you for your review @krys2fa. Much appreciated

**Pull request merged**
